File: Algorithms/74. Search a 2D Matrix/golang/solution.go
func searchMatrix(matrix [][]int, target int) bool {
	m := len(matrix)
	if m == 0 {
		return false
	}
	n := len(matrix[0])

	for left, right := 0, m*n-1; left <= right; {
		pivotIdx := (left + right) / 2
		pivotElement := matrix[pivotIdx/n][pivotIdx%n]
		if pivotElement == target {
			return true
		} else if target < pivotElement {
			right = pivotIdx - 1
		} else {
			left = pivotIdx + 1
		}
	}
	return false
}

File: Algorithms/74. Search a 2D Matrix/php/solution.php
<?php
class Solution
{
    /**
     * @param Integer[][] $matrix
     * @param Integer $target
     * @return Boolean
     */
    function searchMatrix($matrix, $target)
    {
        $m = count($matrix);
        if ($m == 0) {
            return false;
        }
        $n = count($matrix[0]);

        $left = 0;
        $right = $m * $n - 1;
        while ($left <= $right) {
            $pivotIdx = (int)(($left + $right) / 2);
            $pivotElement = $matrix[(int)($pivotIdx / $n)][$pivotIdx % $n];
            if ($pivotElement == $target) {
                return true;
            } else if ($pivotElement < $target) {
                $left = $pivotIdx + 1;
            } else {
                $right = $pivotIdx - 1;
            }
        }
        return false;
    }
}
